### PR TITLE
[MS Exchange Server] Added fields to ignore if empty.

### PR DESCRIPTION
**Pull request by matthewscherer**
## Proposed commit message
A customer came across a couple log lines with missing networkmessageid and senderaddress fields. This in turn rejected the logs messages. Allowing these two fields to be empty or otherwise missing will allow them to pass the ingest function. 

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist
- [ ] 

## Related issues

- Relates https://github.com/elastic/sdh-beats/issues/5552

### CODE REVIEW

**Comment by andrewkroh** [comment]
> changes lgtm but it requires manifest.yml version bump and the changelog.yml

My recommended way to do this is to use

```
elastic-package changelog add \
--link https://github.com/elastic/integrations/pull/12417 \
--type bugfix \
--next patch \
--description 'Handle events where `networkmessageid` or `senderaddress` are not present.'
```

File: packages/microsoft_exchange_server/data_stream/messagetracking/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -51,6 +51,7 @@ processors:
 - set:
     field: email.local_id
     copy_from: microsoft.exchange.networkmessageid
+    ignore_empty_value: true
 - append:
     field: email.sender.address
     value: "{{{microsoft.exchange.senderaddress}}}"
```
Comment (andrewkroh) [inline packages/microsoft_exchange_server/data_stream/messagetracking/elasticsearch/ingest_pipeline/default.yml:57]: According to ECS, [email.sender.address](https://www.elastic.co/guide/en/ecs/current/ecs-email.html#field-email-sender-address) should be a scalar. So I think we should have

```yaml
- set:
    field: email.sender.address
    copy_from: microsoft.exchange.senderaddress
    ignore_empty_value: true
```

This also handles the case where senderaddress is empty.

File: packages/microsoft_exchange_server/data_stream/messagetracking/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -71,6 +72,7 @@ processors:
     ignore_failure: true
```
Comment (andrewkroh) [inline packages/microsoft_exchange_server/data_stream/messagetracking/elasticsearch/ingest_pipeline/default.yml:72]: The above append processor needs to handle the case where senderaddress is emtpy. The current `if` condition looks wrong.

```yaml
- append:
    field: email.from.address
    value: "{{{microsoft.exchange.senderaddress}}}"
    if: ctx.microsoft?.exchange?.senderaddress != null && ctx.microsoft.exchange.senderaddress != ""
```

**Review by andrewkroh — COMMENTED** [review]
This change needs an accompanying test case where a new log sample{s} with empty senderaddress/networkmessageid is added to a packages/microsoft_exchange_server/data_stream/messagetracking/_dev/test/pipeline/*.log file.

**Comment by matthewscherer** [comment]
This PR was replaced with this one. https://github.com/elastic/integrations/pull/12846
